ENH / set up unsupported module versions more tightly

dnsl48 opened the pull request:
related https://github.com/silverstripeltd/open-sourcerers/issues/65

dnsl48 committed:
ENH / set up unsupported module versions more tightly

maxime-rainville commented on composer.json:
```diff
@@ -16,24 +16,24 @@
         "silverstripe/recipe-content-blocks": "^2.3",
         "silverstripe/crontask": "^2.1.2",
-        "silverstripe/dynamodb": "^4.0.1",
+        "silverstripe/dynamodb": "~4.0.1",
```
"silverstripe/dynamodb" might has well be a supported module.

dnsl48 replied:
Ping @chillu 
I also reckon dynamodb must be a supported module, but it's not just yet :)

chillu replied:
That's irreleavnt to this pull request, right? I think a constraint to `~4.0` would be more appropriate. I think we can assume that module sticks to semver

maxime-rainville replied:
FYI `~4.0` is functionality equivalent to `^4.0.0`.

`~4.0.0` will let you upgrade to `4.0.1` but not `4.1.0`. `~4.0` and `^4.0.0` will both upgrade you to `4.1.0`

maxime-rainville commented on composer.json:
```diff
@@ -16,24 +16,24 @@
-        "vulcandigital/silverstripe-seo": "^1.0",
-        "silverstripe/raygun": "^3.1",
+        "vulcandigital/silverstripe-seo": "~1.0.3",
+        "silverstripe/raygun": "~3.1.0",
```
"silverstripe/raygun" and "sminnee/tagmanager" are not officially supported, but I think it's safe to assume that we can trust the maintainers to stick to semver.

chillu replied:
Yeah this is overkill, let's pull in minor releases automatically. There's no point getting notified via the new `composer outdated` if there's `raygun:3.2.0` available. I want to get notified when `raygun:4` is available. Minor releases should be pulled in as part of a `composer update` when it builds the image regularly.

maxime-rainville reviewed (changes requested):
I would keep `silverstripe/dynamodb` with a loose constraint.

The other ones I'm not too fuss about.

dnsl48 commented:
@maxime-rainville I've added the discussed changes, also added a fix for `bin/create-slack-message.php` so it only notifies about stable major upgrades
